server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -2935,10 +2936,24 @@ public void updateVisibleCheckpointForShard(final String allocationId, final Rep
      * @return {@link Tuple} V1 - TimeValue in ms - mean replication lag for this primary to its entire group,
      * V2 - Set of {@link SegmentReplicationShardStats} per shard in this primary's replication group.
      */
-    public Set<SegmentReplicationShardStats> getReplicationStats() {
+    public Set<SegmentReplicationShardStats> getReplicationStatsForTrackedReplicas() {
         return replicationTracker.getSegmentReplicationStats();
     }
 
+    public ReplicationStats getReplicationStats() {
+        if (indexSettings.isSegRepEnabled()) {
```

Comment:
if this is invoked on a replica, lets return null?

Reply:
@Poojita-Raj Thinking this through, it would be safer to return an empty ReplicationStats() object to avoid NPEs.  Lets also check the routing before we return so we aren't unnecessarily computing stats we know aren't there.